src/cpp/aiebu/src/common/utils.h

```
@@ -144,11 +145,47 @@ inline std::string get_pagelabel(const std::string& label)
 // Custom stream buffer that reads from a vector<char>
 class vector_streambuf : public std::streambuf {
 public:
-    vector_streambuf(const std::vector<char>& vec) {
+    explicit vector_streambuf(const std::vector<char>& vec) {
         char* begin = const_cast<char*>(vec.data());
         this->setg(begin, begin, begin + vec.size());
     }
 };
 
+namespace utilities {
+inline std::vector<const char*>
+vector_of_string_to_vector_of_char(const std::vector<std::string>& args)
+{
+  std::vector<const char*> char_vec;
+  for (auto& arg : args)
+    char_vec.push_back(arg.c_str());
+
+  return char_vec;
+}
+}
+
+enum class fragment{
+  BEGIN_ANCHOR_RE,
+  END_ANCHOR_RE,
+  HEX_RE,
+  L_BRACK_RE,
+  R_BRACK_RE,
+};
+
+std::regex get_regex(const std::vector<fragment>& pattern);
+
+constexpr unsigned hexbase = 0x10;
+
+template <typename IntType>
+IntType
+to_uinteger(const std::string& token) {
+  const unsigned long long result = (std::is_same<IntType, uint64_t>::value) ? std::stoull(token, nullptr, 0) :
```

should we put stoul and stoull in try catch ? as they throw invalid_argument or out_of_range exception

---

> should we put stoul and stoull in try catch ? as they throw invalid_argument or out_of_range exception

Only catch if you want to handle the error.  If stoul throws, it is better to propagate the error up to main().